lib/SILOptimizer/Analysis/EscapeAnalysis.cpp

```diff
@@ -101,6 +101,49 @@ EscapeAnalysis::findRecursivePointerKind(SILType Ty,
   return NoPointer;
 }
 
+// Return the PointerKind that summarized an object's stored properties.
+//
+// If a class only holds fields of non-pointer types, then it is guaranteed not
+// to point to any other objects.
+EscapeAnalysis::PointerKind
+EscapeAnalysis::findObjectKind(SILType Ty, const SILFunction &F) const {
+  if (Ty.isAddress())
+    return AnyPointer;
+
+  auto *classDecl = Ty.getClassOrBoundGenericClass();
+  if (!classDecl)
```

Comment:
Why not handle structs and tuples as well? That's a pretty common case.

Reply:
Structs and tuples are handles by getPointerKind. References are always pointers in that case. This specifically looks at object properties.